### PR TITLE
Add port range possibility for "hz-port"

### PR DESCRIPTION
**Pull request by leszko**
- Add feature to make specifying port range (e.g. "5701-5708") for the "hz-port" property (backward compatible change)
- Change default value of "hz-port" from "5701" to "5701-5708"
- Refactor/format changed code
- Add unit tests for the AwsDiscoveryStrategy class
- Update README.md with the change

Background: this change is needed for AWS ECS to work properly, because in case of containers we need to allow multiple Hazelcast instances per one EC2 Instance.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -127,7 +127,7 @@ Following are example declarative and programmatic configuration snippets:
         properties.put("security-group-name","hazelcast");
         properties.put("tag-key","aws-test-cluster");
         properties.put("tag-value","cluster1");
-        properties.put("hz-port","5701");
+        properties.put(hz-port,"5701-5708");
```
Comment (googlielmo) [inline README.md]: looks like key is missing quotes ("hz-port")

File: src/main/java/com/hazelcast/aws/PortRange.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        if (fromPort < MIN_PORT || fromPort > MAX_PORT) {
+            throw new IllegalArgumentException(
+                    String.format("Specified port (%s) outside of port range (%s-%s)", fromPort, MIN_PORT, MAX_PORT));
+        }
+        if (toPort < MIN_PORT || toPort > MAX_PORT) {
+            throw new IllegalArgumentException(
+                    String.format("Specified port (%s) outside of port range (%s-%s)", toPort, MIN_PORT, MAX_PORT));
+        }
+        if (fromPort > toPort) {
+            throw new IllegalArgumentException(String.format("Port %s is greater then %s", fromPort, toPort));
```
Comment (googlielmo) [inline src/main/java/com/hazelcast/aws/PortRange.java]: typo: then -> than

File: src/test/java/com/hazelcast/aws/AwsDiscoveryStrategyFactoryTest.java [hunk 3]
```diff
@@ -90,7 +90,7 @@ public void testFull() throws Exception {
         props.put("tag-key", "test-value");
         props.put("tag-value", "test-value");
         props.put("connection-timeout-seconds", 10);
-        props.put("hz-port", 1234);
+        props.put("hz-port", "1234");
```
Comment (googlielmo) [inline src/test/java/com/hazelcast/aws/AwsDiscoveryStrategyFactoryTest.java]: just wondering if this breaks backward compatibility (for code like this, where someone used an integer for a single port and now a string is required)... maybe worth adding a test to clarify?

Comment (leszko) [reply src/test/java/com/hazelcast/aws/AwsDiscoveryStrategyFactoryTest.java]: That is a super valid comment. Actually, it was not backwards compatible! I've changed it (which made the code slightly worse, but keeps the backward compatibility. Thanks a lot for this comment :)

**Pull request merged**